### ISSUE TITLE
[FEAT][ELEMENTWISE] implement dropout

### ISSUE THREAD

**Issue by lcy-seso** [tile-ai/TileOPs#444]
## 1. Feature Scope

Dropout op with random mask generation. Different kernel pattern from standard elementwise — requires PRNG state management. Separate file `dropout.py`.

- `dropout(x, p, training, seed)` → masked and scaled output
- Deterministic replay: same seed → same mask
- Compatible with PyTorch's dropout semantics: `output = x * mask / (1 - p)` where mask is Bernoulli(1-p)

### Files

| File | Action |
|------|--------|
| `tileops/kernels/dropout.py` | New — DropoutKernel |
| `tileops/ops/dropout.py` | New — DropoutOp |
| `tileops/kernels/__init__.py` | Re-export |
| `tileops/ops/__init__.py` | Re-export |
| `tests/ops/test_dropout.py` | New |

### PRNG Design

Use Philox counter-based RNG (same as PyTorch CUDA):
- State: `(seed, offset)` — deterministic given same pair
- Per-element: `philox4x32(seed, offset + flat_idx)` → 4 random uint32 → convert to uniform float → threshold against `p`
- Each thread generates its own random numbers from deterministic state

### Kernel Sketch

```python
def _make_dropout_kernel(N, dtype, p, seed, offset, threads=256, npt=8):
    block_size = threads * npt
    scale = 1.0 / (1.0 - p)

    @T.prim_func
    def main(x: T.Tensor((N,), dtype), y: T.Tensor((N,), dtype)):
        with T.Kernel(T.ceildiv(N, block_size), threads=threads) as bx:
            for i, j in T.Parallel(threads, npt):
                idx = (bx * threads + i) * npt + j
                # Generate random value from Philox PRNG
                rand_val = philox_uniform(seed, offset + idx)
                mask = rand_val > p
                y[idx] = T.if_then_else(mask, x[idx] * scale, 0.0)
    return main
```

**Open question**: TileLang Philox support — may need to implement Philox rounds manually or use TileLang's random API if available.

### Correctness (3 cases)

| Test | Method |
|------|--------|
| Statistical distribution | Generate mask 100x, verify drop rate within 3σ of target p |
| Deterministic replay | Same seed → bitwise identical output |
| Scale factor | Non-dropped elements scaled by `1/(1-p)` |

### Edge Cases

| Case | Expected |
|------|----------|
| p=0.0 | Output = input (no dropout) |
| p=1.0 | Output = all zeros |
| training=False | Output = input (passthrough) |

## 4. Risks

| Risk | Mitigation |
|------|------------|
| TileLang may lack Philox PRNG primitives | Implement Philox 4-round counter manually in TileLang; or use `T.call_extern` for curand |
| Philox manual implementation correctness | Validate against PyTorch's `torch.cuda.manual_seed` + `F.dropout` output |
| Performance: per-element PRNG may be slow | Benchmark against PyTorch native; Philox is designed for GPU parallelism |

## 5. Acceptance Criteria

- [ ] DropoutKernel + DropoutOp implemented
- [ ] Deterministic replay verified (same seed = same output)
- [ ] Statistical drop rate within 3σ for p ∈ {0.1, 0.3, 0.5}
- [ ] Scale factor correct: non-dropped elements ×(1/(1-p))
- [ ] Edge cases pass: p=0, p=1, training=False
- [ ] `__init__.py` re-exports updated
